lib/asciidoctor/chart/block_macro_processor.rb

```diff
@@ -8,6 +8,7 @@ class BlockMacroProcessor < ::Asciidoctor::Extensions::BlockMacroProcessor
       name_positional_attributes 'type', 'width', 'height', 'axis-x-label', 'axis-y-label', 'data-names'
 
       def process parent, target, attrs
+        parent.document.extend(ChartBlockTracker)
```

Comment:
It would be much better to do this in a preprocessor. That's you're chance to initialize things before any other extension code is called.

Reply:
Indeed, and I will be able to remove `return if instance.instance_variable_defined? :@x_chart` because `extended` will be called only once.

Reply:
Precisely.